cleanrepo/Program.cs

```diff
@@ -794,6 +796,13 @@ private static Dictionary<string, int> GetIncludeFiles(string inputDirectory)
 
         foreach (DirectoryInfo subDirectory in dir.EnumerateDirectories("*", SearchOption.AllDirectories))
         {
+            // If subdirectory pattern is specified, check for this string in the directory path.
+            if (!string.IsNullOrEmpty(subdirectoryPattern) &&
+                !subDirectory.FullName.Contains(subdirectoryPattern, StringComparison.OrdinalIgnoreCase))
```

Comment:
I don't have a strong opinion, but here's the pattern matching alternative:

```suggestion
            if (subdirectoryPattern is string pattern &&
                !subDirectory.FullName.Contains(pattern, StringComparison.OrdinalIgnoreCase))
```